amundsen_application/api/utils/notification_utils.py

```diff
@@ -66,6 +74,12 @@ def send_notification(*, notification_type: str, options: Dict, recipients: List
         return make_response(jsonify({'msg': message}), HTTPStatus.INTERNAL_SERVER_ERROR)
 
 
+@action_logging
+def _send_notification(*, notification_type: str, options: Dict, recipients: List, sender: str) -> None:
```

Comment:
I would recommend renaming to `_log_send_notification` to be more clear. If you look at line 42, it looks like this function sends the notification. Also you can embed this function inside `send_notification` as a private function.